assets/model_monitoring/components/src/action_analyzer/contracts/action.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Action Class."""
+
+import datetime
+import uuid
+import json
+from enum import Enum
+from abc import ABC, abstractmethod
+from action_analyzer.utils.utils import convert_to_camel_case
+
+class ActionSample:
+    """Action sample class."""
+
+    def __init__(self,
+                 question: str,
+                 answer: str,
+                 lookup_score: float,
+                 debugging_info: str,
```

Comment:
is this the DebuggingInfo object? looks like some of the top level parameters is also there such as query_type and top_k

Reply:
For base class, always add the DebuggingInfo (the complex span tree). For the child action sample class, extract some specific parameters based on the need. For example, index action, extract the query_type and top_k to the top level